### ISSUE TITLE
CA1018 (Mark attributes with AttributeUsageAttribute) should ignore abstract attributes

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#1161]
#### Analyzer package

Example: Microsoft.CodeQuality.Analyzers

#### Analyzer

CA1018 (Mark attributes with AttributeUsageAttribute)

#### Repro steps

1. Declare a class which extends `Attribute` and is marked `abstract`

```
public abstract class SomethingBaseAttribute : Attribute
{
  protected SomethingBaseAttribute() { }
}
```

#### Expected behavior

CA1018 should not be reported.

#### Actual behavior

Unknown; the special case is not covered by the current tests for this analyzer.

**Comment by sharwell** [comment]
One more case to add for testing

```csharp
[AttributeUsage(AttributeTargets.Class)]
public abstract class SomethingBaseAttribute : Attribute
{
}

public sealed class SomethingAttribute : SomethingBaseAttribute
{
}
```

:memo: Since `AttributeUsageAttribute` itself is inherited, CA1018 should not be reported in this case.
